linera-core/src/client/mod.rs

```diff
@@ -451,23 +451,27 @@ impl<Env: Environment> Client<Env> {
         certificates: Vec<ConfirmedBlockCertificate>,
     ) -> Result<Option<Box<ChainInfo>>, ChainClientError> {
         let mut info = None;
-        for certificate in certificates {
-            let certificate = Box::new(certificate);
-            let mut result = self.handle_certificate(certificate.clone()).await;
+        let required_blob_ids: Vec<_> = certificates
+            .iter()
+            .flat_map(|certificate| certificate.value().required_blob_ids())
+            .collect();
 
-            if let Err(LocalNodeError::BlobsNotFound(blob_ids)) = &result {
-                let blobs =
-                    futures::stream::iter(blob_ids.iter().copied().map(|blob_id| async move {
+        match self.local_node.read_blob_states_from_storage(&required_blob_ids).await {
+            Ok(_) => (),
+            Err(LocalNodeError::BlobsNotFound(blob_ids)) => {
```

Comment:
I'd prefer to make sure `self.handle_certificate` below returns exactly this error. Why does it currently return only a part of the blob IDs? We should have this check in the chain worker, when processing a confirmed certificate.